buzsaki_lab_to_nwb/original-code/convert_yuta.py

```diff
@@ -221,29 +218,109 @@ def yuta2nwb(session_path='/Users/bendichter/Desktop/Buzsaki/SenzaiBuzsaki2017/Y
     ns.add_position_data(nwbfile, session_path)
 
     shank_channels = ns.get_shank_channels(session_path)[:8]
+    nshanks = len(shank_channels)
     all_shank_channels = np.concatenate(shank_channels)
 
     print('setting up electrodes...', end='', flush=True)
     hilus_csv_path = os.path.join(fpath_base, 'early_session_hilus_chans.csv')
     lfp_channel = get_reference_elec(subject_xls, hilus_csv_path, session_start_time, session_id, b=b)
-    print(lfp_channel)
+    
+    
     custom_column = [{'name': 'theta_reference',
                       'description': 'this electrode was used to calculate LFP canonical bands',
                       'data': all_shank_channels == lfp_channel}]
     ns.write_electrode_table(nwbfile, session_path, custom_columns=custom_column, max_shanks=max_shanks)
+    
+    print('reading raw electrode data...', end='', flush = True)
+    if stub:
+        # example recording extractor for fast testing
+        num_channels = 4
+        num_frames = 10000
+        X = np.random.normal(0, 1, (num_channels, num_frames))
+        geom = np.random.normal(0, 1, (num_channels, 2))
+        X = (X * 100).astype(int)
+        sre = se.NumpyRecordingExtractor(timeseries=X, sampling_frequency=20000, geom=geom)
+    else:
+        nre = se.NeuroscopeRecordingExtractor('{}/{}.dat'.format(session_path,session_id))
+        sre = se.SubRecordingExtractor(nre,channel_ids=all_shank_channels)
+        
+    print('writing raw electrode data...', end='', flush=True)
+    se.NwbRecordingExtractor.add_electrical_series(sre,nwbfile)
+    print('done.')
+    
+    print('reading spiking units...', end='', flush=True)
+    if not stub:
+        spike_times = [200, 300, 400]
+        num_frames = 10000
+        allshanks = []
+        for k in range(nshanks):
+            SX = se.NumpySortingExtractor()
+            for j in range(len(spike_times)):
+                SX.add_unit(unit_id=j+1, times=np.sort(np.random.uniform(0, num_frames, spike_times[j])))
+            allshanks.append(SX)
+        se_allshanks = se.MultiSortingExtractor(allshanks)
+        se_allshanks.set_sampling_frequency(20000)
```

Comment:
Where are you getting 20000?

Reply:
20,000 is the the non-lfp sampling rate that is used to rescale spike times from the `.res` files in both the `to_nwb.neuroscope` `read_spike_times(..., fs=20000.)` and in the `NeuroscopeSortingExtractor` (which much be set by `.set_sampling_frequency()`).

This particular version of the code did not read in the `.XML` in the main conversion code, nor did it have an auxiliary function for retrieving that parameter from the `.XML`, so I hard-coded it similar to the default value in the `to_nwb.neuroscope` `read_spike_times(..., fs=20000.)`; that said, all the future refactors beyond this retrieve and reference the appropriate sampling rate as a variable.

Reply:
That said, if you would like me to undo that hard-code just say the word, wouldn't take that long to do.

Reply:
I went ahead and added this as a parameter pulled from the `.XML` in this script.